tests/e2e/ilab_rhoai_pipeline_runs_test.go

```diff
@@ -0,0 +1,133 @@
+	TrainEffectiveBatchSizePhase1   int     `yaml:"train_effective_batch_size_phase_1" json:"train_effective_batch_size_phase_1"`
+	TrainEffectiveBatchSizePhase2   int     `yaml:"train_effective_batch_size_phase_2" json:"train_effective_batch_size_phase_2"`
+	TrainLearningRatePhase1         float64 `yaml:"train_learning_rate_phase_1" json:"train_learning_rate_phase_1"`
+	TrainLearningRatePhase2         float64 `yaml:"train_learning_rate_phase_2" json:"train_learning_rate_phase_2"`
+	TrainMaxBatchLen                int     `yaml:"train_max_batch_len" json:"train_max_batch_len"`
+	TrainNNodes                     int     `yaml:"train_nnodes" json:"train_nnodes"`
+	TrainNProcPerNode               int     `yaml:"train_nproc_per_node" json:"train_nproc_per_node"`
+	TrainNumEpochsPhase1            int     `yaml:"train_num_epochs_phase_1" json:"train_num_epochs_phase_1"`
+	TrainNumEpochsPhase2            int     `yaml:"train_num_epochs_phase_2" json:"train_num_epochs_phase_2"`
+	TrainNumWarmupStepsPhase1       int     `yaml:"train_num_warmup_steps_phase_1" json:"train_num_warmup_steps_phase_1"`
+	TrainNumWarmupStepsPhase2       int     `yaml:"train_num_warmup_steps_phase_2" json:"train_num_warmup_steps_phase_2"`
+	TrainSaveSamples                int     `yaml:"train_save_samples" json:"train_save_samples"`
+	TrainSeed                       int     `yaml:"train_seed" json:"train_seed"`
+}
+
+func loadPipelineParams(filename string) (*PipelineParams, error) {
+	workingDirectory, err := os.Getwd()
+	if err != nil {
+		return nil, fmt.Errorf("error getting working directory: %w", err)
+	}
+
+	filepath := filepath.Join(workingDirectory, filename)
+	data, err := os.ReadFile(filepath)
+	if err != nil {
+		return nil, fmt.Errorf("error reading file %s: %w", filename, err)
+	}
+
+	var params PipelineParams
+	err = yaml.Unmarshal(data, &params)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshaling YAML from file %s: %w", filename, err)
+	}
+	return &params, nil
+}
+func TestPipelineRun(t *testing.T) {
+
+	if os.Getenv("ENABLE_ILAB_PIPELINE_TEST") != "true" {
+		t.Skip("Skipping iLab pipeline test. Set ENABLE_ILAB_PIPELINE_TEST=true to enable.")
+	}
+
+	pipelineServerURL := os.Getenv("PIPELINE_SERVER_URL")
+	require.NotEmpty(t, pipelineServerURL, "PIPELINE_SERVER_URL environment variable must be set")
+
+	bearerToken := os.Getenv("BEARER_TOKEN")
+	require.NotEmpty(t, bearerToken, "BEARER_TOKEN environment variable must be set")
+
+	pipelineDisplayName := os.Getenv("PIPELINE_DISPLAY_NAME")
+	require.NotEmpty(t, pipelineDisplayName, "PIPELINE_DISPLAY_NAME environment variable must be set")
+
+	// Retrieve the pipeline ID
+	pipelineID, err := TestUtil.RetrievePipelineId(t, pipelineServerURL, pipelineDisplayName, bearerToken)
+	require.NoError(t, err, "Failed to retrieve pipeline ID")
+
+	// Load input parameters for the pipeline
+	parameters, err := loadPipelineParams("../e2e/resources/pipeline_params.yaml")
```

Comment:
you can get rid of a lot of this load params custom logic if we just use viper: 
https://github.com/spf13/viper

for example you can do something like: 

```golang
	viper.SetConfigName("pipeline_params")
	viper.SetConfigType("yaml")
	viper.AddConfigPath("../e2e/resources/")
	err := viper.ReadInConfig()
	require.NoError(t, err)
	paramsMap := viper.AllSettings()
```